src/main.rs

```diff
@@ -458,6 +502,24 @@ async fn main() {
         }
     }
 
+    // metrics secret create once and use by all pods
+    if kub_controller.metrics.is_some() {
+        let metrics_secret = match kub_controller.create_metrics_secret() {
+            Ok(secret) => secret,
+            Err(err) => {
+                error!("Failed to create metrics secret! {err}");
+                return;
+            }
+        };
```

Comment:
nit: if you have the main function call a function returning a  `Result<(), Box<dyn std::error::Error>>`, and printing the error, most of these can be reduced with `?`s, ie
```suggestion
        let metrics_secret = kub_controller.create_metrics_secret()?;
        info!("something good happened"!);
```